File: 57_01_TwoNumbersWithSum/TwoNumbersWithSum.py
class Solution:
    def FindNumbersWithSum(self, array, tsum):
        # write code here
        res = []
        i, j = 0, len(array) - 1
        while i < j:
            s = array[i] + array[j]
            if s < tsum:
                i += 1
            elif s > tsum:
                j -= 1
            else:
                res.append(array[i])
                res.append(array[j])
                break
        return res

File: 57_01_TwoNumbersWithSum/TwoNumbersWithSum.cpp
#include <iostream>
#include <vector>
#include <string>
#include <deque>
#include <set>
#include <map>
#include <queue>
#include <stack>
#include <algorithm>
using namespace std;

class Solution
{
public:
    vector<int> FindNumbersWithSum(vector<int> array, int sum)
    {
        vector<int> res;
        int i = 0, j = array.size() - 1;
        // int product = 1e9;
        while (i < j)
        {
            int s = array[i] + array[j];
            if (s == sum)
            {
                // if (array[i] * array[j] < product)
                // {
                // product = array[i] * array[j];
                res.push_back(array[i]);
                res.push_back(array[j]);
                break;
                // }
            }
            else if (s < sum)
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return res;
    }
};